xdsl/irdl/declarative_assembly_format_parser.py

```diff
@@ -259,12 +259,14 @@ class _AttrExtractor(VarExtractor[ParsingState]):
         name: str
         is_prop: bool
         inner: VarExtractor[Attribute]
+        default_value: Attribute | None
```

Comment:
what are cases where this is `None`?

Reply:
Worth a comment?

Reply:
Sure

Reply:
Wrote a comment, thought about it for 10 minutes, deleted it and wrote a different comment. If it looks OK I'll merge